content/docs/standard-library/core/_index.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Core
+weight: 5
+---
+The `Core`  section describes the core component's.
+
+# Emitter
+The `Emitter` constantly emits a message that you bind to it using the compiler directive `#bind`.This should be avoided if possible (use const's and literals instead).
+```
+pub component Emitter<T>() (msg T)
+```
+
+### Usage
+
+# Destructor
+```
+pub component Destructor(msg any) ()
+```
+
+# Blocker
+Blocker works like an AND gate, only sending data through the outport once it receives data at both inports.
+```
+pub component Blocker<T>(sig any, data T) (data T)
+```
+
+# Struct Builder
+The `builder` is used to construct your custom struct types.
```

Comment:
You can add that compiler generate inports for every instance of this component based on given type argument which must be struct. Inports relates to structs field. Component blocks until values for all fields provided and then sends `msg` to outport.